extensions/procfs/tests/ProcFsMonitorTests.cpp

```diff
@@ -0,0 +1,157 @@
+      CHECK_FALSE(document.HasMember("Network"));
+      CHECK_FALSE(document.HasMember("Process"));
+      CHECK(document.HasMember("Memory"));
+    }
+    sleep(1);
+    {
+      const auto& result = test_controller_.trigger();
+
+      REQUIRE(result.at(ProcFsMonitor::Success).size() == 1);
+      const auto& result_flow_file = result.at(ProcFsMonitor::Success)[0];
+
+      rapidjson::Document document;
+      auto content = test_controller_.plan->getContent(result_flow_file);
+      document.Parse(content.c_str());
+      REQUIRE(document.IsObject());
+      CHECK(document.HasMember("CPU"));
+      CHECK(document.HasMember("Disk"));
+      CHECK(document.HasMember("Network"));
+      CHECK(document.HasMember("Process"));
+      CHECK(document.HasMember("Memory"));
+    }
+  }
+
+  SECTION("Relative OpenTelemetry") {
+    test_controller_.plan->setProperty(proc_fs_monitor, ProcFsMonitor::ResultRelativenessProperty.getName(), "Relative");
+    test_controller_.plan->setProperty(proc_fs_monitor, ProcFsMonitor::OutputFormatProperty.getName(), "OpenTelemetry");
+    {
+      const auto& result = test_controller_.trigger();
+
+      REQUIRE(result.at(ProcFsMonitor::Success).size() == 1);
+      const auto& result_flow_file = result.at(ProcFsMonitor::Success)[0];
+
+      rapidjson::Document document;
+      auto content = test_controller_.plan->getContent(result_flow_file);
+      document.Parse(content.c_str());
+      REQUIRE(document.IsObject());
+      REQUIRE(document.HasMember("Body"));
+      // First trigger has not enough information for relative output
+      CHECK_FALSE(document["Body"].HasMember("CPU"));
+      CHECK_FALSE(document["Body"].HasMember("Disk"));
+      CHECK_FALSE(document["Body"].HasMember("Network"));
+      CHECK_FALSE(document["Body"].HasMember("Process"));
+      CHECK(document["Body"].HasMember("Memory"));
+    }
+    sleep(1);
+    {
+      const auto& result = test_controller_.trigger();
+
+      REQUIRE(result.at(ProcFsMonitor::Success).size() == 1);
+      const auto& result_flow_file = result.at(ProcFsMonitor::Success)[0];
+
+      rapidjson::Document document;
+      auto content = test_controller_.plan->getContent(result_flow_file);
+      document.Parse(content.c_str());
+      REQUIRE(document.IsObject());
+      REQUIRE(document.HasMember("Body"));
+      CHECK(document["Body"].HasMember("CPU"));
+      CHECK(document["Body"].HasMember("Disk"));
+      CHECK(document["Body"].HasMember("Network"));
+      CHECK(document["Body"].HasMember("Process"));
+      CHECK(document["Body"].HasMember("Memory"));
+    }
+  }
+
+  SECTION("Relative without wait") {
+    test_controller_.plan->setProperty(proc_fs_monitor, ProcFsMonitor::ResultRelativenessProperty.getName(), "Relative");
+    const auto& result1 = test_controller_.trigger();
+    REQUIRE(result1.at(ProcFsMonitor::Success).size() == 1);
+    const auto& result2 = test_controller_.trigger();
+    REQUIRE(result2.at(ProcFsMonitor::Success).size() == 1);
+  }
```

Comment:
I don't think "Relative without wait" is a useful test case, as the total elapsed CPU time can be zero, and we divide by it in `SerializeNormalizedCPUStat()` (as happened on the last CI run).  I would remove this test case.

Reply:
On the contrary. I made this test so it makes sure we dont divide by zero. It turns out I removed a check during the last refactor.
I've fixed the issue https://github.com/apache/nifi-minifi-cpp/pull/1152/commits/6248890ce7c068ee8fa4eab52598b18a933b388c, I think we can keep this.